### ISSUE TITLE
[iOS][tvOS] System.Net.Requests.Tests Process terminated due to System.IO.DirectoryNotFoundException

### ISSUE THREAD

**Issue by mdh1418** [dotnet/runtime#52117]
The test suite `System.Net.Requests.Tests` crashes on iOS/tvOS

```
info	13:27:25.265137-0400	System.Net.Requests.Tests	((null) warning) Process terminated due to "Certificates cctor threw System.IO.DirectoryNotFoundException: Could not find a part of the path '/Users/mdhwang/Library/Developer/CoreSimulator/Devices/71D00C72-A1BD-4EF7-87EA-B22842EC1650/data/Containers/Bundle/Application/964D9EE9-013C-4685-BF81-3E021F818A5D/System.Net.Requests.Tests.app/TestDataCertificates/testservereku.contoso.com.pfx'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter) in System.Private.CoreLib.dll:token 0x600000a+0x48
   at Interop.CheckIo(Error error, String path, Boolean isDirectory, Func`2 errorRewriter) in System.Private.CoreLib.dll:token 0x600000b+0x10
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) in System.Private.CoreLib.dll:token 0x60000a1+0x62
   at System.IO.Strategies.FileStreamHelpers.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options) in System.Private.CoreLib.dll:token 0x600560d+0xc
   at System.IO.Strategies.Net5CompatFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) in System.Private.CoreLib.dll:token 0x600561b+0x43
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) in System.Private.CoreLib.dll:token 0x600560c+0x0
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) in System.Private.CoreLib.dll:token 0x6005608+0x0
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) in System.Private.CoreLib.dll:token 0x6005338+0x17f
   at System.IO.File.ReadAllBytes(String path) in System.IO.FileSystem.dll:token 0x60000fe+0x1
   at System.Net.Test.Common.Configuration.Certificates..cctor() in System.Net.Requests.Tests.dll:token 0x6000217+0x3e
   at System.Diagnostics.DefaultTraceListener.Fail(String message, String detailMessage) in System.Diagnostics.TraceSource.dll:token 0x6000040+0x0
   at System.Diagnostics.DefaultTraceListener.Fail(String message) in System.Diagnostics.TraceSource.dll:token 0x600003f+0x0
   at System.Diagnostics.TraceInternal.Fail(String message) in System.Diagnostics.TraceSource.dll:token 0x60000b5+0x0
   at System.Diagnostics.Trace.Fail(String message) in System.Diagnostics.TraceSource.dll:token 0x600007d+0x0
   at System.Net.Test.Common.Configuration.Certificates..cctor() in System.Net.Requests.Tests.dll:token 0x6000217+0x0
   at System.Net.Test.Common.LoopbackServer.Connection.CreateAsync(Socket socket, Stream stream, Options httpOptions) in System.Net.Requests.Tests.dll:token 0x600024f+0x0
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[<CreateAsync>d__13](<CreateAsync>d__13& stateMachine) in System.Private.CoreLib.dll:token 0x600464a+0x0
   at System.Net.Test.Common.LoopbackServer.Connection.CreateAsync(Socket socket, Stream stream, Options httpOptions) in System.Net.Requests.Tests.dll:token 0x600024f+0x0
   at System.Net.Test.Common.LoopbackServer.EstablishConnectionAsync() in System.Net.Requests.Tests.dll:token 0x6000078+0x0
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1[[System.Net.Test.Common.LoopbackServer.Connection, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51],[System.Net.Test.Common.LoopbackServer.<EstablishConnectionAsync>d__15, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]].ExecutionContextCallback(Object s) in System.Private.CoreLib.dll:token 0x6004675+0x0
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in System.Private.CoreLib.dll:token 0x6002b1e+0x0
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1[[System.Net.Test.Common.LoopbackServer.Connection, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51],[System.Net.Test.Common.LoopbackServer.<EstablishConnectionAsync>d__15, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]].MoveNext(Thread threadPoolThread) in System.Private.CoreLib.dll:token 0x6004679+0x0
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1[[System.Net.Test.Common.LoopbackServer.Connection, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51],[System.Net.Test.Common.LoopbackServer.<EstablishConnectionAsync>d__15, System.Net.Requests.Tests, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]].MoveNext() in System.Private.CoreLib.dll:token 0x6004678+0x0
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining) in System.Private.CoreLib.dll:token 0x600309a+0x0
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in System.Private.CoreLib.dll:token 0x6002fac+0x0
   at System.Threading.Tasks.Task.FinishContinuations() in System.Private.CoreLib.dll:token 0x6002fab+0x0
   at System.Threading.Tasks.Task`1[[System.Net.Sockets.Socket, System.Net.Sockets, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].TrySetResult(Socket result) in System.Private.CoreLib.dll:token 0x6002e8b+0x0
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.Net.Sockets.Socket, System.Net.Sockets, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].SetExistingTaskResult(Task`1 task, Socket result) in System.Private.CoreLib.dll:token 0x600466d+0x0
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.Net.Sockets.Socket, System.Net.Sockets, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].SetResult(Socket result) in System.Private.CoreLib.dll:token 0x600466c+0x0
   at System.Net.Sockets.Socket.CompleteAccept(Socket s, TaskSocketAsyncEventArgs`1 saea) in System.Net.Sockets.dll:token 0x6000284+0x0
   at System.Net.Sockets.Socket.<>c.<AcceptAsync>b__268_0(Object s, SocketAsyncEventArgs e) in System.Net.Sockets.dll:token 0x60002bf+0x0
   at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted(SocketAsyncEventArgs e) in System.Net.Sockets.dll:token 0x60002df+0x0
   at System.Net.Sockets.SocketAsyncEventArgs.OnCompletedInternal() in System.Net.Sockets.dll:token 0x60002de+0x0
   at System.Net.Sockets.SocketAsyncEventArgs.FinishOperationAsyncSuccess(Int32 bytesTransferred, SocketFlags flags) in System.Net.Sockets.dll:token 0x6000307+0x0
   at System.Net.Sockets.SocketAsyncEventArgs.CompletionCallback(Int32 bytesTransferred, SocketFlags flags, SocketError socketError) in System.Net.Sockets.dll:token 0x6000326+0x0
   at System.Net.Sockets.SocketAsyncEventArgs.AcceptCompletionCallback(IntPtr acceptedFileDescriptor, Byte[] socketAddress, Int32 socketAddressSize, SocketError socketError) in System.Net.Sockets.dll:token 0x600030e+0x0
   at System.Net.Sockets.SocketAsyncContext.AcceptOperation.InvokeCallback(Boolean allowPooling) in System.Net.Sockets.dll:token 0x600045d+0x0
   at System.Net.Sockets.SocketAsyncContext.OperationQueue`1[[System.Net.Sockets.SocketAsyncContext.ReadOperation, System.Net.Sockets, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].ProcessAsyncOperation(ReadOperation op) in System.Net.Sockets.dll:token 0x6000472+0x0
   at System.Net.Sockets.SocketAsyncContext.ProcessAsyncReadOperation(ReadOperation op) in System.Net.Sockets.dll:token 0x60003f8+0x0
   at System.Net.Sockets.SocketAsyncContext.ReadOperation.System.Threading.IThreadPoolWorkItem.Execute() in System.Net.Sockets.dll:token 0x6000430+0x0
   at System.Net.Sockets.SocketAsyncContext.AsyncOperation.Process() in System.Net.Sockets.dll:token 0x6000426+0x0
   at System.Net.Sockets.SocketAsyncContext.HandleEvents(SocketEvents events) in System.Net.Sockets.dll:token 0x600041a+0x0
   at System.Net.Sockets.SocketAsyncEngine.System.Threading.IThreadPoolWorkItem.Execute() in System.Net.Sockets.dll:token 0x6000482+0x0
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in System.Private.CoreLib.dll:token 0x6002c8a+0x0
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in System.Private.CoreLib.dll:token 0x6002db5+0x0
   at System.Threading.Thread.StartCallback() in System.Private.CoreLib.dll:token 0x600297e+0x0
```

**Comment by filipnavara** [comment]
Will be fixed by #52372 + #52191

**Comment by marek-safar** [comment]
@mdh1418 the fixes landed, could you try to reenable the tests

**Comment by filipnavara** [comment]
Still blocked by https://github.com/dotnet/runtime/pull/53029 iirc.

**Comment by mdh1418** [comment]
This suite no longer fails with `System.IO.DirectoryNotFoundException`, so I'm closing this issue, but it now fails with another issue https://github.com/dotnet/runtime/issues/56621
